AsyncDisplayKit/Private/ASInternalHelpers.h

```diff
@@ -33,6 +33,26 @@ BOOL ASRunningOnOS7();
 ASDISPLAYNODE_EXTERN_C_END
 
 /**
+ @summary Conditionally performs UIView geometry changes in the given block without animation and call completion block afterwards.
+ 
+ Used primarily to circumvent UITableView forcing insertion animations when explicitly told not to via
+ `UITableViewRowAnimationNone`. More info: https://github.com/facebook/AsyncDisplayKit/pull/445
+ 
+ @param withoutAnimation Set to `YES` to perform given block without animation
+ @param block Perform UIView geometry changes within the passed block
+ @param completion Call completion block if UIView geometry changes within the passed block did complete
+ */
+ASDISPLAYNODE_INLINE void ASPerformBlockWithoutAnimationCompletion(BOOL withoutAnimation, void (^block)(), void (^completion)()) {
```

Comment:
One question, which I'm not sure of, is if the UIKit Version has any other Side effects. I would hope it is a pure convenience for the Core animation method, but it also seems possible that it could affect how the table internally performs operations

That said, I am fine with us changing to this approach now that I better understand it. However, referencing my comments on the first revision it seems like the best approach to support the need to check for batch fetching on every edited operation could be to have a custom replacement for this method that is specific to table and collection and handles calling the Batch check API.

It would be a good idea to introduce a protocol, Like the batch fetching scroll view, to expose this kind of API and allow a shared piece of code to rely on it. It may make sense to make the shared protocol something more general, such as ASRangeView or something, as a precursor to the larger project to create a shared superclass for the table and collection node. The protocol might allow us to start creating shared method names that can be used externally I objects that are referencing only a pointer that conforms to that protocol rather than a table or collection specifically. This protocol stuff is not required for the first version of this change, although it may be very quick to use such an approach to ensure the editing methods are able to disable animation and schedule a batch call in one convenient line rather than eight separate call points